### PR TITLE
#1703 Patient history modal

### LINKED ISSUE #1703: Patient History Modal
## Is your feature request related to a problem? Please describe.

Need a way to view a patients history

## Describe the solution you'd like

A modal which allows viewing of a patients history

## Implementation

N/A

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #1703 

## Change summary

- Adds `PatientHistory` modal
- Does not include any 'repeats' logic, just a list of trans lines for a patient currently.

image

## Testing

N/A

### Related areas to think about

N/A

### CODE REVIEW

File: src/database/DataTypes/TransactionBatch.js [hunk 1]
```diff
@@ -35,6 +35,18 @@ export class TransactionBatch extends Realm.Object {
...
+  /**
+   * Returns either the prescriber name who prescribed the medicine
+   * to a patient, or an empty string.
+   * @return {String}
+   */
+  get prescriber() {
+    const { prescriber } = this.transaction;
+    const { firstName = '', lastName = '' } = prescriber || {};
+    if (!firstName && !lastName) return '';
+    return `${firstName} ${lastName}`;
```
Comment (mark-prins) [inline src/database/DataTypes/TransactionBatch.js]: perhaps a `trim` to cater for the *very* unlikely noFirstName scenario:

```suggestion
    return `${firstName} ${lastName}`.trim();
```

File: src/pages/dataTableUtilities/getColumns.js [hunk 2]
```diff
@@ -225,13 +233,21 @@ const COLUMNS = () => ({
   },
 
   // STRING COLUMNS
-
+  [COLUMN_NAMES]: {
```
Comment (mark-prins) [inline src/pages/dataTableUtilities/getColumns.js]: Should this be?:

```suggestion
  [COLUMN_NAMES.PRESCRIBER]: {
```

File: src/widgets/modals/PatientHistory.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        onPress={onSortColumn}
+        isAscending={isAscending}
+        sortKey={sortKey}
+      />
+    ),
+    [sortKey, isAscending]
+  );
+
+  const getCallback = React.useCallback(colKey => {
+    switch (colKey) {
```
Comment (mark-prins) [inline src/widgets/modals/PatientHistory.js]: `case 'comingSoon'`?

File: src/pages/dataTableUtilities/getColumns.js [hunk 4]
```diff
@@ -182,6 +183,12 @@ const PAGE_COLUMNS = {
   ],
   prescriberSelect: [COLUMN_NAMES.FIRST_NAME, COLUMN_NAMES.LAST_NAME],
   itemSelect: [COLUMN_NAMES.CODE, COLUMN_NAMES.NAME],
+  patientHistory: [
+    COLUMN_NAMES.ITEM_CODE,
+    COLUMN_NAMES.ITEM_NAME,
+    COLUMN_NAMES.TOTAL_QUANTITY,
+    'prescriber',
```
Comment (mark-prins) [inline src/pages/dataTableUtilities/getColumns.js]:
```suggestion
    COLUMN_NAMES.PRESCRIBER,
```

File: src/widgets/modals/PatientHistory.js (hunk 3 above)
Comment (joshxg) [reply src/widgets/modals/PatientHistory.js]: yeah - will be handling the `REPEAT` column

File: src/pages/dataTableUtilities/getColumns.js (hunk 4 above)
Comment (joshxg) [reply src/pages/dataTableUtilities/getColumns.js]: whoops thanks! e900d57838e74655a9238c5be7e6b0eb25f7dadc

File: src/pages/dataTableUtilities/getColumns.js (hunk 2 above)
Comment (joshxg) [reply src/pages/dataTableUtilities/getColumns.js]: yup e900d57838e74655a9238c5be7e6b0eb25f7dadc

File: src/database/DataTypes/TransactionBatch.js (hunk 1 above)
Comment (joshxg) [reply src/database/DataTypes/TransactionBatch.js]: nice! be6c4307f5d8db21ef83011666bcd7fb47892445

**Pull request merged**